src/util/mongo.controller.js

```diff
@@ -62,6 +62,29 @@ const updateResource = async (id, data, resourceType) => {
   return { id: results.value.id, created: false };
 };
 
+/**
+ * searches for a document and updates it by pushing to existing if found, creates it if not
+ * @param {*} id id of resource to be updated
+ * @param {*} data the new data to push in the document
+ * @param {*} resourceType the collection the document is in
+ * @returns the id of the updated/created document
+ */
+const pushToResource = async (id, data, resourceType) => {
+  const collection = db.collection(resourceType);
+
+  const results = await collection.findOneAndUpdate({ id: id }, { $push: data });
+
+  // If the document cannot be created with the passed id, Mongo will throw an error
+  // before here, so should be ok to just return the passed id
+  if (results.value === null) {
+    // null value indicates a newly created document
+    return { id: id, created: true };
+  }
+
+  // value being present indicates an update, so set created flag to false
+  return { id: results.value.id, created: false };
+};
```

Comment:
Do we need this block of code? Since we're not doing any upserting, I don't think we'll ever reach this `if` since it will only be updating existing documents.

We also don't make use of the return value at all, so I think it's okay for this function to be void and just do the `findOneAndUpdate` with no return value

Reply:
Good call! I think my initial conception was that this could still work for a database document that doesn't exist yet, but that actually fails, so I'm going to make these changes and add that comment to the function docs.